### PR TITLE
LPA-3561

### PR DESCRIPTION
**Pull request by andrewpearce-digital**
## Purpose
Create a script to update database credentials

Fixes LPA-3561

## Approach

Script the configuration of cloud9
Script the ALTER USER sql action
Script the update procedure including an ecs redeploy

## Learning

_Any tips and tricks, blog posts or tools which helped you. Plus anything notable you've discovered about the Refunds service_

## Checklist

* [x] I have performed a self-review of my own code
* [x] I have updated documentation (Confluence/GitHub wiki/tech debt doc) where relevant
* [x] I have added tests to prove my work
* [x] The product team have tested these changes

### CODE REVIEW

File: docs/runbooks/cloud9/cloud9_init.sh [hunk 1]
```diff
@@ -1,45 +1,95 @@
...
+function add_rds_sgs() {
+  local current_sg_name
+  local current_sg_id
+  local environment_name=$(echo ${1:?} | awk '{print tolower($0)}')
+  local applications_rds_client_sg
+  local caseworker_rds_client_sg
+  local CURRENT_SG_IDS=()
+
+  echo "getting current sg names and instance id..."
+  current_sg_names=$(curl -s http://169.254.169.254/latest/meta-data/security-groups)
```
Comment (williamfalconeruk) [inline docs/runbooks/cloud9/cloud9_init.sh]: Are these IP addresses configurable or are they a known static address?

Comment (andrewpearce-digital) [reply docs/runbooks/cloud9/cloud9_init.sh]: It's static, this is the aws metadata service https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/instancedata-data-retrieval.html

File: docs/runbooks/rotating_database_credentials/README.md [hunk 2]
```diff
@@ -25,3 +25,34 @@ cd ~/environment/opg-refunds/docs/runbooks/rotating_database_credentials
...
+
+git clone https://github.com/ministryofjustice/opg-refunds.git
+cd ~/environment/opg-refunds/
+git checkout LPA-3561
+
+. ~/environment/opg-refunds/docs/runbooks/cloud9/cloud9_init.sh preproduction
+
+~/environment/opg-refunds/docs/runbooks/maintenance_mode/manage_maintenance.sh \
+  --environment preproduction \
+  --maintenace_mode
```
Comment (williamfalconeruk) [inline docs/runbooks/rotating_database_credentials/README.md]: typo...

File: docs/runbooks/rotating_database_credentials/README.md [hunk 3]
```diff
@@ -25,3 +25,34 @@ cd ~/environment/opg-refunds/docs/runbooks/rotating_database_credentials
...
+  --credential applications_write \
+  --username applications
+
+
+echo "Here we're doing some manual testing of refunds..."
+sleep 20
+
+~/environment/opg-refunds/docs/runbooks/maintenance_mode/manage_maintenance.sh \
+  --environment preproduction \
+  --disable_maintenace_mode
```
Comment (williamfalconeruk) [inline docs/runbooks/rotating_database_credentials/README.md]: typo in maintenance

File: docs/runbooks/maintenance_mode/manage_maintenance.sh [hunk 4]
```diff
@@ -57,7 +57,5 @@ function start() {
 
 MAINTENACE_MODE=False
```
Comment (williamfalconeruk) [inline docs/runbooks/maintenance_mode/manage_maintenance.sh]: should this be MAINTENANCE_MODE?

File: docs/runbooks/maintenance_mode/manage_maintenance.sh [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+function parse_args() {
+  for arg in "$@"
+  do
+      case $arg in
+          -e|--environment)
+          ENVIRONMENT="$2"
+          shift
+          shift
+          ;;
+          -m|--maintenace_mode)
```
Comment (williamfalconeruk) [inline docs/runbooks/maintenance_mode/manage_maintenance.sh]: should this be maintenance_mode? I've spotted a few of these - just checking...

File: docs/runbooks/maintenance_mode/manage_maintenance.sh [hunk 6]
```diff
@@ -34,28 +34,28 @@ function parse_args() {
...
+  if [ $MAINTENANCE_MODE = "True" ]
   then
     enable_maintenance
   else
     disable_maintenance
   fi
 }
 
-MAINTENACE_MODE=False
+MAINTENAnCE_MODE=False
```
Comment (williamfalconeruk) [inline docs/runbooks/maintenance_mode/manage_maintenance.sh]: I'm being very picky- the case is mixed here...

Comment (andrewpearce-digital) [reply docs/runbooks/maintenance_mode/manage_maintenance.sh]: image

**Pull request merged**